### DISCUSSION TITLE
Pass a variable to create_file based on another template?

### DISCUSSION THREAD

**Discussion by DeutscheGabanna** [SilentVoid13/Templater#872] [Help]
```
tp.file.create_new(tp.file.find_tfile("new meeting"), final_name, true, app.vault.getAbstractFileByPath("meetings"))
```

How to pass a variable to that newly created file based on another template?

**Comment by AB1908** [comment]
Can you demonstrate the workflow with an example?

**Reply by DeutscheGabanna** [reply]
I work mostly within daily notes. Before a meeting starts, I go to my daily note and trigger "new meeting reference" template.
What it does is create and embed a new "side" note specifically for that meeting. It follows a structure of `with <person> about <topic> - YYYY-MM-DD`.

The "new meeting reference" template after creating that blank note and renaming it hands over the workflow to that new meeting note. And so, "new meeting" takes control and pastes its standard frontmatter.

I could rely solely on `tp.date.now("YYYY-MM-DD")` to put the date the meeting was held into the frontmatter. But, sometimes, I add meeting notes to old dailies (based on haphazard logs), where `tp.date.now()` fails.

Meanwhile, I found a solution to my problem. Since "new meeting reference" already includes the date in the note's title:
```
// notice that dates don't match up
if (held != triggered_date) {
	if (await tp.system.prompt("You triggered this template in a note on " + triggered_date + ". Do you want to use it instead of today's date? y / ESC", false, false, false) == "y") held = triggered_date
}

// arrange the name of the note following the <person> + <topic> + <date> structure
final_name += topic + " - " + held

// embed, create based on another template and hand over the rest of the work to that new file
tR += "!\[\[" + final_name + "\]\]"
tp.file.create_new(tp.file.find_tfile("new meeting"), final_name, true, app.vault.getAbstractFileByPath("meetings"))
```

So I can reference that in the new note like so:
```
---
tags: type/meeting
held: <% tp.date.now("YYYY-MM-DD", 0, tp.file.title, "YYYY-MM-DD") %>
---
```

**Reply by DeutscheGabanna** [reply]
Looks like this:
![image](https://user-images.githubusercontent.com/59067099/196266379-cfc044d5-7a00-4f2e-b0cc-fe9d44f31458.png)
